CommonTools/PileupAlgos/plugins/PuppiProducer.cc

```diff
@@ -351,10 +351,55 @@ void PuppiProducer::beginJob() {}
 void PuppiProducer::endJob() {}
 // ------------------------------------------------------------------------------------------
 void PuppiProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
-  //The following says we do not know what parameters are allowed so do no validation
-  // Please change this to state exactly what you do use, even if it is no parameters
   edm::ParameterSetDescription desc;
-  desc.setUnknown();
+  desc.add<bool>("puppiDiagnostics", false);
+  desc.add<bool>("puppiForLeptons", false);
+  desc.add<bool>("UseDeltaZCut", true);
+  desc.add<double>("DeltaZCut", 0.3);
+  desc.add<double>("PtMaxNeutrals", 200.);
+  desc.add<bool>("useExistingWeights", false);
+  desc.add<bool>("useWeightsNoLep", false);
+  desc.add<bool>("clonePackedCands", false);
+  desc.add<int>("vtxNdofCut", 4);
+  desc.add<double>("vtxZCut", 24);
+  desc.add<edm::InputTag>("candName", edm::InputTag("particleFlow"));
+  desc.add<edm::InputTag>("vertexName", edm::InputTag("offlinePrimaryVertices"));
+  desc.add<bool>("applyCHS", true);
+  desc.add<bool>("invertPuppi", false);
+  desc.add<bool>("useExp", false);
+  desc.add<double>("MinPuppiWeight", .01);
+
+  edm::ParameterSetDescription puppialgos;
+  puppialgos.add<int>("algoId", 5);
+  puppialgos.add<bool>("useCharged", false);
+  puppialgos.add<bool>("applyLowPUCorr", false);
+  puppialgos.add<int>("combOpt", 5);
+  puppialgos.add<double>("cone", .4);
+  puppialgos.add<double>("rmsPtMin", .1);
+  puppialgos.add<double>("rmsScaleFactor", 1.0);
+
+  edm::ParameterSetDescription algos;
+  std::vector<double> etaMin;
```

Comment:
Defining uninitialized vectors is useless. You could just define a single dummy vector, e.g.
```C++
std::vector<double> dummy_vdouble;
```
and use it to ```add``` everywhere.

Or (probaly better?) you can fill with some reasonable value, e.g. the ones from CommonTools/PileupAlgos/python/Puppi_cff.py:
```C++
  algos.add<std::vector<double>>("etaMin", {0.});
  algos.add<std::vector<double>>("etaMax", {2.5});
  algos.add<std::vector<double>>("ptMin", {0.});
  algos.add<std::vector<double>>("MinNeutralPt", {0.2});
```
etc.

Reply:
By the way: this default algo configuration corresponds to what? I couldn't find it in the actual config in cmssw